integration/shareddata/benchmark.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package shareddata provides data for tests and benchmarks.
+package shareddata
+
+import "go.mongodb.org/mongo-driver/bson"
+
+// SimpleBenchmarkValues contain documents with various types of values.
+var SimpleBenchmarkValues BenchmarkValues = NewBenchmarkValues(
+	"Ha59TmC9nwTXqEob73HIwqWUcx2Lp7aengCtNsqjWK0=",
```

Comment:
I'm not sure what this hash is for. The `GenerateMixedValues()` generates values so hash should be different right?

Reply:
((discussed on meeting but leaving comments to write things on paper))

The hash here is the expected hash of all documents fetched from BenchmarkValues's iterator. There are two reasons behind the checksum:

1. The documents in BenchmarkProvider (which is the interface that is implemented by BenchmarkValues) must be exactly the same every time with the same order. That's why each provider has expected hash of all the documents. Setup function, while fetching the data from provider calculates the actual checksum and compares both expected and actual hash to be sure that we generate data properly.

2. When we compare results of the benchmarks with tooling like `benchstat`, test cases are compared with each other by testname. In the scenario, where we change the documents in provider, we don't want to compare the old results (on old data) with the new results (on new data). That's why we append the checksum of provider to the testname.

Reply:
Thanks for this. 
Comment about hash would be helpful, also my personal preference is to call `seed` instead of `generate` :hugs: but maybe like I said, it's personal :see_no_evil:

Reply:
I believe that term seeding is much more commonly used as the starting point of pseudo random values chain. Here we want to make clear that data provided cannot be random.

Reply:
I agree with the second reason, but I'm not sure about the first one. Storing expected hashes seems to be overcomplicated. After all, hash changes are immediately noticeable anyway – you run benchmark twice without any changes, and you should get the same hashes

Reply:
I'm forward with the idea of not storing hash here, and running benchmark twice to output the hash! It would be simpler

Reply:
I'm not even sure we want to run the benchmark twice. As long as the data's hash is a part of the benchmark name, we will notice if it changes.

Reply:
> After all, hash changes are immediately noticeable anyway – you run benchmark twice without any changes, and you should get the same hashes

Not sure how is it noticeable. If the hash changed, the new test case with the new hash in name will be created. The benchstat will just notice that the new test case is created and won't compare it to anything. There's no place where we validate that hash is changed or not, and it would need to be checked manually.

Reply:
So we will check that manually